common/views.py

```diff
@@ -205,8 +206,33 @@ def post(self, request, *args, **kwargs):
         if user_form_details.is_valid() and profile_form_details.is_valid():
             try:
                 with transaction.atomic():
-                    user_form_details.save()
-                    profile_form_details.save()
+                    user_form_retrived_data = user_form_details.save(commit=False)
+
+                    # Only saving the new values, overwriting other existing values
+                    if (
+                        user_form_retrived_data.first_name != user_profile_data.user.first_name
+                        or user_form_retrived_data.last_name != user_profile_data.user.last_name
+                    ):
+                        user_form_retrived_data.username = user_profile_data.user.username
+                        user_form_retrived_data.email = user_profile_data.user.email
+                        user_form_retrived_data.password1 = user_profile_data.user.password
+                        user_form_retrived_data.password2 = user_profile_data.user.password
```

Comment:
I don't fully get it, why are we doing this?

Reply:
I wanted to save only the new information. I removed it, as I checked the the previous way is okay to prevent the curl injection.